File: frontend/systeme-interne/src/app/models/PlanDeValidationModel/plandevalidationmodule.ts
import { Plandevalidationmoduleid } from './plandevalidationmoduleid';
import { Plandevalidation } from './plandevalidation'; // Assuming you have already defined PlanDeValidation
import { Element } from './element';

export class Plandevalidationmodule {
  id!: Plandevalidationmoduleid;
  cne!: string;
  moduleID!: number;
  moduleIntitule?:string;
  semestreID!: string;
  filiereID!: number;
  filiereNom?:string
  planDeValidationsElements?: Plandevalidation[];
  noteFinale?: number;
  resultat?: string; // Assuming Resultat is a string enum in your Spring Boot project
  elements?:Element[]
}

File: backend/Deliberation-session-ordinaire-service/src/main/java/com/example/deliberationsessionordinaireservice/entities/PlanDeValidationModule.java
package com.example.deliberationsessionordinaireservice.entities;

import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Getter
@Setter
@IdClass(PlanDeValidationModuleID.class)
public class PlanDeValidationModule implements Serializable {
    @Id
    private String CNE;
    @Id
    private Long moduleID;
    @Id
    private String semestreID;
    @Id
    private Long filiereID;

    @Transient
    private List<PlanDeValidation> planDeValidationsElements;

    private Float noteFinale;

    @Enumerated(EnumType.STRING)
    private Resultat resultat;

}
